README.md

```diff
@@ -1,2 +1,68 @@
-# code-review
-Example code for the code review
+# Https client
+
+- [https-client](#https-client)
+  - [Introduction](#introduction)
+  - [Pre-requisites](#pre-requisites)
+  - [Getting Started](#getting-started)
+
```

Comment:
Would be nice to have some examples here on how to use the lib. If we provide a lib to someone (especially an opensource), we would like people to use it, and examples help understanding and better adoption as a consequence.